src/order/compactly_generated.lean

```diff
@@ -301,10 +316,34 @@ theorem complete_lattice.independent_iff_finite {s : set α} :
     exact ⟨ha, set.subset.trans ht (set.diff_subset _ _)⟩ }
 end⟩
 
+lemma complete_lattice.independent_Union_of_directed {η : Type*}
+  {s : η → set α} (hs : directed (⊆) s)
+  (h : ∀ i, complete_lattice.independent (s i)) :
+  complete_lattice.independent (⋃ i, s i) :=
+begin
+  by_cases hη : nonempty η,
+  { resetI,
+    rw complete_lattice.independent_iff_finite,
+    intros t ht,
+    rcases set.finite_subset_Union t.finite_to_set ht with ⟨I, fi, hI⟩,
+    rcases hs.finset_le fi.to_finset with ⟨i, hi⟩,
+    exact (h i).mono (set.subset.trans hI $ set.bUnion_subset $
+      λ j hj, hi j (set.finite.mem_to_finset.2 hj)) },
+  { refine complete_lattice.independent_empty.mono _,
+    rintro _ ⟨_, ⟨i, _⟩, _⟩, exact hη ⟨i⟩ }
```

Comment:
The following works without invoking `independent_empty.mono`, but it doesn't really matter.
```suggestion
  { rintros a ⟨_, ⟨i, _⟩, _⟩,
    exfalso, exact hη ⟨i⟩, },
```